src/filestore.rs

```diff
@@ -0,0 +1,224 @@
+use std::{
+    collections::BTreeMap,
+    fs,
+    path::{Path, PathBuf},
+};
+
+use colored::Colorize;
+use serde::Deserialize;
+
+use crate::{
+    model::{Annotation, GPUSpecRequirement, KeyInfo, Pod, ToYaml},
+    store::Store,
+};
+
+static POD_FOLDER_NAME: &str = "Pod";
+
+pub struct FileStore {
+    storage_folder_path: PathBuf,
+}
+
+impl FileStore {
+    pub fn new(storage_folder_path: PathBuf) -> Self {
```

Comment:
let's make it more generic and take in anything that can be turned into PathBuf. So more appropriate will be 
```rust
pub fn new(storage_folder_path: impl Into<PathBuf>) -> Self
```

Reply:
I actually still need to look up the correct way to pass around path. It seems that in std they use something like this:
```pub fn some_public_func<P: AsRef<Path>>(path: P) -> ... {
    some_public_func_imp(path.as_ref())
}

fn _some_public_func(path: &Path) -> ... { ... }```